File: c++/478. Generate Random Point in a Circle.py
class Solution(object):

    def __init__(self, radius, x_center, y_center):
        """
        :type radius: float
        :type x_center: float
        :type y_center: float
        """
        self.radius = radius
        self.x_center = x_center
        self.y_center = y_center

    def randPoint(self):
        """
        :rtype: List[float]
        """
        rad = random.random()*math.pi*2
        r = random.random()*self.radius*self.radius
        r = math.sqrt(r)
        return [self.x_center + r*math.cos(rad), 
            self.y_center + r*math.sin(rad)]
        


# Your Solution object will be instantiated and called as such:
# obj = Solution(radius, x_center, y_center)
# param_1 = obj.randPoint()

File: go/src/leetcode/478/478. Generate Random Point in a Circle.go
/**
@author ZhengHao Lou
Date    2022/06/05
*/
package main

import (
	"math"
	"math/rand"
)

type Solution struct {
	r, x, y float64
}

func Constructor(radius float64, x_center float64, y_center float64) Solution {
	return Solution{
		r: radius,
		x: x_center,
		y: y_center,
	}
}

func (this *Solution) RandPoint() []float64 {
	theta := math.Pi * 2 * rand.Float64()
	r := math.Sqrt(rand.Float64() * this.r * this.r)
	return []float64{math.Sin(theta)*r + this.x, math.Cos(theta)*r + this.y}
}
